src/shared/components/person/settings.tsx

```diff
@@ -1474,6 +1544,120 @@ export class Settings extends Component<any, SettingsState> {
     }
   }
 
+  handleImportFileChange(i: Settings, event) {
+    i.setState({ settingsFile: event.target.files?.item(0) });
+  }
+
+  async handleExportSettings(i: Settings) {
+    i.setState({ importOrExportLoading: "exporting" });
+    const res = await HttpService.client.exportSettings();
+
+    if (res.state === "success") {
+      i.exportSettingsLink.current!.href = encodeURI(
+        `data:application/json,${JSON.stringify(res.data)}`,
+      );
+      i.exportSettingsLink.current?.click();
+    } else if (res.state === "failed") {
+      toast(
+        res.err.message === "rate_limit_error"
+          ? "Please wait a bit before trying to export or import again"
+          : "Error exporting settings",
+        "danger",
+      );
+    }
+
+    i.setState({ importOrExportLoading: null });
+  }
+
+  async handleImportSettings(i: Settings) {
+    i.setState({ importOrExportLoading: "importing" });
+
+    const res = await HttpService.client.importSettings(
+      JSON.parse(await i.state.settingsFile!.text()),
+    );
+
+    if (res.state === "success") {
+      toast("Settings successfully imported!", "success");
```

Comment:
Its probably a good idea to just reload the page a few seconds after the successful import, rather than doing this below. Its a pretty rare operation anyway. If this works below tho, its pry fine.